src/app/qgssettingstree.cpp

```diff
@@ -71,6 +71,22 @@ QgsSettingsTree::QgsSettingsTree( QWidget *parent )
   keyIcon.addPixmap( style()->standardPixmap( QStyle::SP_FileIcon ) );
 
   connect( &refreshTimer, &QTimer::timeout, this, &QgsSettingsTree::maybeRefresh );
+  connect( this, &QTreeWidget::itemClicked, this, &QgsSettingsTree::itemClicked );
+}
+
+void QgsSettingsTree::itemClicked( QTreeWidgetItem *item, int column )
+{
+  Q_UNUSED( column );
+  if ( !item )
+  {
+    return;
+  }
+
+  QVariant value = item->data( 2, Qt::UserRole );
```

Comment:
Doesn't calling  `setEditTriggers( QAbstractItemView::AllEditTriggers );`  do the same thing?

Reply:
@nyalldawson , setEditTriggers(...) only works if you click on the editable column. This approach enables the editor irrespective of the column.

It's partly because I'm trying to get rid of the glitch whereas the checkbox is painted over the text.